cow_instrument/DetectorInfo.h

```diff
@@ -40,27 +40,45 @@ template <typename InstTree> class DetectorInfo {
 
   bool isMonitor(size_t detectorIndex) const;
 
-  void initL2();
+  double l2(size_t detectorIndex) const;
 
-  void initL1();
+  Eigen::Vector3d position(size_t componentIndex) const;
 
-  double l2(size_t detectorIndex) const;
+  Eigen::Vector3d positionDetector(size_t detectorIndex) const;
```

Comment:
`DetectorInfo` provides accessors for detector *and* components. This dual-purpose interface (including two types of indices) seems a bit confusing. Is there a way to do this differently?